mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -93,7 +106,12 @@ private[regression] trait GeneralizedLinearRegressionBase extends PredictorParam
         Family.fromName($(family)) -> Link.fromName($(link))), "Generalized Linear Regression " +
         s"with ${$(family)} family does not support ${$(link)} link function.")
     }
-    super.validateAndTransformSchema(schema, fitting, featuresDataType)
+    val newSchema = super.validateAndTransformSchema(schema, fitting, featuresDataType)
```

Comment:
We do not assume ```predictionCol``` is always set. If we set ```predictionCol = ""```, we will not output ```predictionCol```. We checked ```colName.isEmpty``` at [here](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala#L90).